### PR TITLE
fix: Remove hardcoded texts (WIP)

### LINKED ISSUE #4164: Core/Creature: hardcoded gossips in zone_witergrasp
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Northrend/zone_wintergrasp.cpp#L34-L37

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701910-core-creature-hardcoded-gossips-in-zone_witergrasp?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4166: Core/Creature: hardcoded gossips in zone_howling_fjord
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Northrend/zone_howling_fjord.cpp#L311-L312

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701908-core-creature-hardcoded-gossips-in-zone_howling_fjord?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4169: Core/Creature: hardcoded gossips in culling_of_stratholme
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/culling_of_stratholme.cpp#L139-L144

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701905-core-creature-hardcoded-gossips-in-culling_of_stratholme?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4176: Core/Creature: hardcoded gossips in zone_tanaris
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Kalimdor/zone_tanaris.cpp#L279
https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Kalimdor/zone_tanaris.cpp#L325-L330

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701898-core-creature-hardcoded-gossips-in-zone_tanaris?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4180: Core/Creature: hardcoded gossips in zone_moonglade
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Kalimdor/zone_moonglade.cpp#L45-L46
https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Kalimdor/zone_moonglade.cpp#L102-L105
https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Kalimdor/zone_moonglade.cpp#L160-L161

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701894-core-creature-hardcoded-gossips-in-zone_moonglade?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4190: Core/Creature: hardcoded gossips in zone_western_plaguelands
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/EasternKingdoms/zone_western_plaguelands.cpp#L32-L37
https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/EasternKingdoms/zone_western_plaguelands.cpp#L111

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701876-core-creature-hardcoded-gossips-in-zone_western_plaguelands?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4191: Core/Creature: hardcoded gossips in zone_undercity
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/EasternKingdoms/zone_undercity.cpp#L304-L306
https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/EasternKingdoms/zone_undercity.cpp#L924-L925

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701875-core-creature-hardcoded-gossips-in-zone_undercity?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4192: Core/Creature: hardcoded gossips in battlegroundav
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/game/Battlegrounds/Zones/BattlegroundAV.h#L12-L14

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701874-core-creature-hardcoded-gossips-in-battlegroundav?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by r-o-b-o-t-o**
## Changes Proposed:
Remove a lot of hardcoded texts in the C++ sources, and use texts from the database instead (gossips, emotes, etc)
See https://github.com/azerothcore/azerothcore-wotlk/projects/15

## Issues Addressed:

- Closes #4192 
- Closes #4191 
- Closes #4190 
- Closes #4180 
- Closes #4176
- Closes #4169
- Closes #4166
- Closes #4164

## How to Test the Changes:

- Alterac Valley (done ✅):
https://github.com/azerothcore/azerothcore-wotlk/issues/4192
Verify that the NPCs (Balinda Stonehearth, Galvangar, Snivvle) yell:
  * "Begone. Uncouth scum! The Alliance shall prevail in Alterac Valley!"
  * "Now is the time to attack! For the Horde!"
  * "Snivvle is here! Snivvle claims the Coldtooth Mine!"

- Undercity, Parqual Fintallas (done ✅):
https://github.com/azerothcore/azerothcore-wotlk/issues/4191
  * `.go c 32022`
  * `.q add 6628`
  * Talk to Parqual Fintallas and verify that the gossip options are like the following: https://youtu.be/UOwwR1yvroQ?t=306

- Western Plaguelands (done ✅):
https://github.com/azerothcore/azerothcore-wotlk/issues/4190
  * `.go c 45238`
  * `.q add 5226`
  * Turn in [Return to Chillwind Camp]
  * Commander Ashlam Valorfist gives a quest, turn it in
  * Talk to Alchemist Arbington
  * Verify that the gossip options are:
     "What's needed for the cauldron at Felstone Field? / Dalson's Tears? / the Writhing Haunt? / Gahrron's Withering?"
     "I need a vitreous focuser."

- Karazhan **(WIP 🔧)**:
https://github.com/azerothcore/azerothcore-wotlk/issues/4188
  * `.go c 135823`

- Zul'Aman **(WIP 🔧)**:
https://github.com/azerothcore/azerothcore-wotlk/issues/4187
  * Halazzi: `.go c 86195`
  * Akil'zon: `.go c 86494`
  * Nalorakk: `.go c 86609`
  * Jan'alai: `.go c 89322`

  * Harkor: [https://youtu.be/ni2MtU7SHvc?t=567](https://youtu.be/ni2MtU7SHvc?t=567)
Well then, I spose I be owin' ya my thanks
We've killed your captors. You're free to go.

  * Tanzar: [https://youtu.be/ni2MtU7SHvc?t=1194](https://youtu.be/ni2MtU7SHvc?t=1194)
You kill the trollbear, mon?
Nalorakk is dead, you're free to go.

  * Kraz: [https://youtu.be/ni2MtU7SHvc?t=1896](https://youtu.be/ni2MtU7SHvc?t=1896)
Freedom! Kraz life was over
We've killed your captors. You're free to go.

  * Ashli: [https://youtu.be/ni2MtU7SHvc?t=2720](https://youtu.be/ni2MtU7SHvc?t=2720)
Oh hooray - you killed the bad troll just in time to save me
It's safe, little gnome. You can come out now.

- Moonglade (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4180
  * "I'd like to fly to Rut'theran Village." `.go c 42338`
  * "I'd like to fly to Thunder Bluff." `.go c 42337`

  * "Do you know where I can find Half Pendant of Aquatic Endurance?" / "Do you know where I can find Half Pendant of Aquatic Agility?"
     Alliance: `.q add 272` 
     Horde: `.q add 30`
     then `.go c 42337` `.go c 42338`

  * "What do you represent, spirit?"
    "I seek to understand the importance of strength of the body."
    "I seek to understand the importance of strength of the heart."
    "I have heard your words, Great Bear Spirit, and I understand. I now seek your blessings to fully learn the way of the Claw."
       * Alliance: `.q add 5929`
       * Horde: `.q add 5930`
       * then `.go c 42356`

- Tanaris (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4176
  First part:
    * `.go c steward_of_time`
    * Take the quest [To The Master's Lair]
    * `.go c 23284` if you don't want to enjoy the ride
    * Turn in [To The Master's Lair]
    * `.go c steward_of_time`
    * Verify that the gossip option is "Please take me to the master's lair."

  Second part:
    * `.q add 2946`
    * `.go -9619 -2815 15 1`
    * Turn in [Seeing What Happens]
    * Take [The Stone Watcher]
    * Verify the gossip options (https://youtu.be/cT4Zv791uHg?t=409):
      * "What function do you serve?"
      * "What are the Plates of Uldum?"
      * "Where are the Plates of Uldum?"
      * "Excuse me? We've been "reschedueled for visitations"? What does that mean?!"
      * "So, what's inside Uldum?"
      * "I will return when i have the Plates of Uldum."

- The Culling of Stratholme (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4169
    * `.tele cullingofstratholme`
    * Find Arthas (`.go c 1970935`)
    * Progress through the instance, and verify that Arthas' gossip options are like the following:
    * "Yes, my Prince. We are ready."
    * "We're only doing what is best for Lordaeron, your Highness."
    * "Lead the way, Prince Arthas."
    * "I'm ready."
    * For Lordaeron!"
    * "I'm ready to battle the dreadlord, sire."

- Howling Fjord (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4166
  `.q add 11221`
  * "High Executor Anselm wants a report on the situation."
    * `.go c 113843`
  * "High Executor Anselm requests your report."
    * `.go c 117636`
    * `.respawn darkrangerlyana` might help if she died fighting

- Wintergrasp (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4164
  * `.tele wintergrasp`
  * `.bf start 1`
  * Join the battle
  * `.aura 55629`
  * `.go 4937 2388 330`
  * Verify that you have the following gossip items:
    * "I'd like to build a catapult."
    * "I'd like to build a demolisher."
    * "I'd like to build a siege engine."

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by r-o-b-o-t-o** [comment]
Just to be clear, I'm posting this PR now so that I can write the test cases as I go, but don't test this now guys, I still have 18 more commits to do 😬

File: src/server/game/Battlegrounds/Zones/BattlegroundAV.cpp [hunk 1]
```diff
@@ -383,24 +383,42 @@ void BattlegroundAV::PostUpdateImpl(uint32 diff)
...
             else
             {
                 if (i == 0)
                 {
                     CastSpellOnTeam(AV_BUFF_A_CAPTAIN, TEAM_ALLIANCE);
                     Creature* creature = GetBGCreature(AV_CPLACE_MAX + 61);
                     if (creature)
-                        YellToAll(creature, LANG_BG_AV_A_CAPTAIN_BUFF, LANG_COMMON);
+                    {
+                        BroadcastText const* broadcastText = sObjectMgr->GetBroadcastText(LANG_BG_AV_A_CAPTAIN_BUFF);
```
Comment (Kitzunu) [inline src/server/game/Battlegrounds/Zones/BattlegroundAV.cpp:401]: Should use creature_text instead

File: data/sql/updates/pending_db_world/rev_1620594734059183600.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+DELETE FROM `gossip_menu_option` WHERE (`MenuID` = 4764) AND (`OptionID` IN (0, 1, 2, 3));
+
+INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
+(1013, 0, 0, 'Kel''Thuzad is my answer.', 8377, 1, 1, 0, 0, 0, 0, '', 0, 0);
+
+INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
+(1013, 1, 0, 'Gul''dan is my answer.', 8378, 1, 1, 0, 0, 0, 0, '', 0, 0);
+
+INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
+(1013, 2, 0, 'Ner''zhul is my answer.', 8380, 1, 1, 0, 0, 0, 0, '', 0, 0);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1620594734059183600.sql:16]:
```suggestion
INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
(1013, 0, 0, 'Kel''Thuzad is my answer.', 8377, 1, 1, 0, 0, 0, 0, '', 0, 0),
(1013, 1, 0, 'Gul''dan is my answer.', 8378, 1, 1, 0, 0, 0, 0, '', 0, 0),
(1013, 2, 0, 'Ner''zhul is my answer.', 8380, 1, 1, 0, 0, 0, 0, '', 0, 0);
```

**Comment by FrancescoBorzi** [comment]
nice improvement!

**Comment by FrancescoBorzi** [comment]
@r-o-b-o-t-o what's the status of this PR?

**Comment by r-o-b-o-t-o** [comment]
> @r-o-b-o-t-o what's the status of this PR?

Still WIP, I kept that in my mind but haven't had the time to work on it since then 😅
If it can be merged without conflicts maybe it's better to merge this one now and I'll do a part 2 later

**Comment by r-o-b-o-t-o** [comment]
I don't think it has been tested by anyone except myself though.

**Comment by Nyeriah** [comment]
Hey,

This seems to be fine to me, I think if we could get this merged it'd be really cool.

I'm working on unifying the gossip scripting API, so if we postpone it we might end up running into merge conflicts that will give us a pretty hard time :P

**Comment by FrancescoBorzi** [comment]
any updates on this?

**Comment by acidmanifesto** [comment]
@r-o-b-o-t-o what is the current status of this pr?

**Comment by r-o-b-o-t-o** [comment]
Feel free to merge it if @Nyeriah still needs it

**Comment by acidmanifesto** [comment]
@r-o-b-o-t-o we have conflict

**Review by Nyeriah — APPROVED** [review]

**Comment by Kitzunu** [comment]
I'd like to merge what we have of this PR. Only problem being that gossip menu is supposed to be sniffed...

**Comment by Kitzunu** [comment]
Merging as it is a good start and too long abandoned

**Comment by Bogir** [comment]
> Merging as it is a good start and too long abandoned

It must be a little fixed
All abstract numbers must be changed to named variable
изображение
Leaving it like this is bad manners.

**Comment by Kitzunu** [comment]
Yea, but the author has abandoned the PR so this is better than what we had before.

**Comment by Kitzunu** [comment]
This should eventually go into the DB anyways =)

**Comment by Bogir** [comment]
I closed my PR and keep testing it. 
https://github.com/azerothcore/azerothcore-wotlk/pull/11017
If I'll do not notice failures - I will open it again.

**Pull request merged**